contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -318,6 +323,34 @@ def get_address_type(self, idx: int):
             return 'ipv6'
         return 'dns'
 
+    def is_tor_only(self):
+        """ Checks if a node has only TOR and no addresses announced """
+        if not self.announced:
+            return False
+        if len(self.addresses) == 0:
+            return False
+        for i in range(len(self.addresses)):
+            if self.get_address_type(i) != 'tor':
+                return False
+        return True
+
+    def is_tor_strict(self):
+        """ Checks if a node is TOR only and does not connect to non-TOR """
+        if not self.is_tor_only():
+            return False
+        for c in self.channels:
+            other = c.node1 if self != c.node1 else c.node2
+            if not other.announced:
+                continue
+            has_tor = False
+            for a in other.addresses:
+                if ".onion:" in a:
+                    has_tor = True
+                    break
```

Comment:
Why this and not using other.is_tor_only?

Reply:
The reason I implemented this without the usage of `is_tor_only` is because the case where `other.announced == False` or `len(other.addresses) == 0`: In this case I skip this node and `continue` with the next other node to see if it has non-tor addresses. If I would re-use `is_tor_only` and do this in the loop:  

```py
            if not other.is_tor_only():
                return False
```

will return too few 'strict' nodes, as `is_tor_only` returns `False` on unannounced nodes or nodes without any addresses, then this would terminate the loop too quickly imho.

What do you think?

Reply:
Or maybe we should add a `def has_clearnet()` and use that inside `is_tor_strict()`

Reply:
I added  `has_tor()` (and `has_clearnet()`) and re-used that inside the `is_tor_strict()` method to get the result I wanted. This way its more readable.